File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/HR/SalaryHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.HR
{
    public class SalaryHistory : AuditableEntity
    {
        public long Id { get; set; }
        public int Year { get; set; }
        public int Month { get; set; }
        public decimal BasicSalary { get; set; }

        // Totals from attendance & transactions
        public decimal TotalBonus { get; set; }
        public decimal TotalOvertime { get; set; }
        public decimal TotalDeductions { get; set; }
        public decimal TotalLoans { get; set; }

        // Final Calculated Salary
        public decimal NetSalary { get; set; }

        public long EmployeeId { get; set; }
        public Employee Employee { get; set; }
    }
}

File: LMS_CMS_Frontend/src/app/Models/HR/salary-history.ts
export class SalaryHistory {
    constructor(
        public id: number = 0,
        public year: number = 0,
        public month: number = 0,

        public basicSalary: number = 0,
        public totalBonus: number = 0,
        public totalOvertime: number = 0,
        public totalDeductions: number = 0,
        public totalLoans: number = 0,
        public netSalary: number = 0,


        public employeeId: number = 0,
        public employeeEnName: string = '',
        public employeeArName: string = '',

        public insertedAt: string = "",
        public insertedByUserId: number = 0,
    ) { }

}


